homeassistant/components/zha/light.py

```diff
@@ -541,3 +567,11 @@ async def async_update(self) -> None:
         # Bitwise-and the supported features with the GroupedLight's features
         # so that we don't break in the future when a new feature is added.
         self._supported_features &= SUPPORT_GROUP_LIGHT
+
+    async def _force_member_updates(self):
+        """Force the update of member entities to ensure the states are correct for bulbs that don't report their state."""
+        component = self.hass.data[light.DOMAIN]
+        entities = [component.get_entity(entity_id) for entity_id in self._entity_ids]
+        tasks = [entity.async_get_state(from_cache=False) for entity in entities]
```

Comment:
The light component shouldn't be used from a platform. We should use the dispatch helper instead to signal the relevant entities.

Reply:
This is where I 🤷🏻‍♂️ Tell me what u want us to do and I’ll do i

Reply:
Sometimes I wish HA had a `f"{domain}. update_state"` service. I don't quite understand understand why this should be any different from `light.turn_on` service

Reply:
Here's an example of how to use the dispatch helper:
https://github.com/home-assistant/core/blob/650d61e4f3007d1f7d456713d43fbc30b7396ce6/homeassistant/components/tuya/__init__.py#L160

https://github.com/home-assistant/core/blob/650d61e4f3007d1f7d456713d43fbc30b7396ce6/homeassistant/components/tuya/__init__.py#L212-L216

In our case we can simply let the light entities that  should be targeted by the forced update subscribe to a common force update signal string. Those entities subscribe in `async_added_to_hass` with the `async_get_state` method as target callback, or if needed an intermediate method.

Reply:
Yeah ZHA uses this everywhere... I’m thinking of how to do this efficiently. The problem we have here is this is a group entity and the entities that make up the group don’t know anything about the group or the group entity. They don’t even know they’re in the group. But the group entity does have the info it needs. We also can’t do anything with entity Id based signals because the entity id can be changed. That’s why I chose to do this like this.

I could signal the group object and then move this code to the ZHAGroup so that it isn’t in light.py would that solve your concern with the platform issue?

Reply:
@MartinHjelmare how's this: https://github.com/dmulcahey/home-assistant/commit/ca37afa12583d5f1f154c5fcf207460103c95dc9 If that's good I'll PR the update

Reply:
Yes, that looks good!